### ISSUE TITLE
The problem of paris traceroute in TCP

### ISSUE THREAD

**Issue by xhdix** [RIPE-NCC/ripe-atlas-probe-busybox#8]
Measurement: [39288974](https://atlas.ripe.net/measurements/39288974/#probes)
Probe: [1002641](https://atlas.ripe.net/frames/probes/1002641/)

In paris traceroute mode, there are usually two duplicate hop:

![image](https://user-images.githubusercontent.com/12384263/160249500-bedd5c5c-5ede-49a8-85a8-2cfa26fc8958.png)

I do not know how this can happen. (I have not read the code yet)

you can see it in all measurement of probe `1002641` on 2022-03-10:

![image](https://user-images.githubusercontent.com/12384263/160249509-a496cc3b-76e5-424e-b793-d1040e845e36.png)

all of which were in paris traceroute mode:

![image](https://user-images.githubusercontent.com/12384263/160249516-dfcf76ce-fdc5-46c4-bb82-335f2a0c83e0.png)

**Comment by PhilipHomburg** [comment]
Random question, is it possible that there are actually two nodes that have the same IP address? Your traceroute has a lot of RFC 1918 address and an IPv4 link local. All of those are not guaranteed to be unique.

**Comment by xhdix** [comment]
I don't think so. There is no same IP address in standard traceroute and RTTs are also the same. (Although, the first RTT is a bit strange.)
Also, they doing static routing.

![image](https://user-images.githubusercontent.com/12384263/160404749-7b4bc426-adc2-4293-bbad-cf600b950c2a.png)

https://atlas.ripe.net/measurements/39537917/#probes

**Comment by PhilipHomburg** [comment]
Ah. I see. It is possible that the packets of two measurements got mixed up.

**Comment by xhdix** [comment]
So this can also happen in standard traceroute?

**Comment by PhilipHomburg** [comment]
Hard to say. Paris traceroute and standard traceroute have a different way of storing parameters in packets
